docs/src/manual/implementing.md

```diff
@@ -511,6 +511,29 @@ These methods have the following rules:
   variable or constraint
 * If there are multiple variables or constraints with the name, throw an error.
 
+!!! warning
+    You should _not_ implement [`ConstraintName`](@ref) for
+    [`SingleVariable`](@ref) constraints. If you implement [`ConstraintName`](@ref)
+    for other constraints, you may need to add the following two methods to
+    avoid method ambiguities:
```

Comment:
I don't think there is any ambiguity issue anymore, is it?

Reply:
I've rephrased. It may still be needed if they implement a `::MOI.ConstraintIndex` method. Ideally, Julia would have a "not-subtype" relation:
```julia
where {F !<: MOI.SingleVariable}
```

Reply:
That would be futuristic